matching_test.go

```diff
@@ -0,0 +1,29 @@
+package backoff
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+)
+
+var (
+	customError    = fmt.Errorf("custom error")
+	differentError = fmt.Errorf("different error")
+
+	tests = map[error]error{
```

Comment:
Opinion, we should try to define these within the scope of the test function itself. That way we do not pollute the package scope.